msteams-platform/messaging-extensions/high-quality-message-extension.md

```diff
@@ -705,25 +733,73 @@ Message extensions respond to a user input with an Adaptive Card. An Adaptive Ca
 
 * If a user can change any information on the card through dialog, Stageview, or directly from the card, we recommend the Adaptive Card to support universal actions and automatic refresh. [*Recommended*]
 * Adaptive Cards must include a URL as part of the [metadata](https://adaptivecards.io/explorer/Metadata.html), which allows cards to be easily copied from one hub to another. [*Recommended*]
 * Apart from thumbnails, any image in an Adaptive Card must have an alt-text. [*Recommended*]
 
+## Best practices to extend Teams apps as plugin in Copilot
+
+If you have a search-based message extension and have enabled the M365 channel, your app works in Copilot for Microsoft 365 in Teams. To enable your message extensions as a plugin in Copilot for Microsoft 365, we recommend you to implement the following best practices:
+
+* **Upgrade TeamsJS Version**: Upgrade your [TeamsJS version to 2.19.0](https://www.npmjs.com/package/@microsoft/teams-js) by installing the `@microsoft/teams-js` package from npm. This ensures that your application continues to work with Bing domains.
+
+* **Update Microsoft Entra ID app registration for SSO**: Microsoft Entra ID single sign-on (SSO) for message extensions works the same way in Bing as it does in Teams or Outlook. However, you need to add Bing’s client application identifiers to your bot's Microsoft Entra ID app registration in your tenant's App registrations portal. To do this, follow these steps:
+
+   1. Go to [Azure portal](https://ms.portal.azure.com/) and sign in with your sandbox tenant account.
+   1. Select **App registrations**.
+   1. Search for your app and select the app name.
+   1. Go to **Manage** > **Expose an API**.
+   1. In the Authorized client applications section, ensure that the following Client ID values are listed:
+
+      | Microsoft 365 client application | Client ID |
+      | --- | --- |
+      | Bing | 9ea1ad79-fdb6-4f9a-8bc3-2b70f96e34c7 |
+      | Bing (Staging) | ef47e344-4bff-4e28-87da-6551a21ffbe0 |
+      |Teams desktop and mobile| 1fec8e78-bce4-4aaf-ab1b-5451cc387264|
+      |Teams web | 5e3ce6c0-2b1f-4285-8d4b-75ee78787346|
+      |Microsoft 365 web | 4765445b-32c6-49b0-83e6-1d93765276ca|
+      |Microsoft 365 desktop | 0ec893e0-5785-4de6-99da-4ed124e5296c|
+      |Microsoft 365 mobile | d3590ed6-52b3-4102-aeff-aad2292ab01c |
+      |Outlook desktop | d3590ed6-52b3-4102-aeff-aad2292ab01c |
+      | Outlook Web Access | bc59ab01-8403-45c6-8796-ac3ef710b3e3 |
+      |Outlook mobile | 27922004-5251-4030-b22d-91ecd9a37ea4 |
+
+* **Configure Content Security Policy Headers**: If your app uses Content Security Policy (CSP) headers, make sure to allow the following frame-ancestors in your CSP headers:
+
+   | Microsoft 365 app host | frame-ancestor permission |
+   | --- | --- |
+   | Bing | edgeservices.bing.com, <www.bing.com>, <www.staging-bing-int.com>, copilot.microsoft.com |
```

Comment:
formatting suggestion, For both Bing & App Copilot, can we be consistent in how we are separating domains, either use a comma or a new line in both rows